common/kvstore/src/test/java/org/apache/spark/util/kvstore/RocksDBSuite.java

```diff
@@ -255,12 +257,15 @@ public void testNegativeIndexValues() throws Exception {
       }
     });
 
-    List<Integer> results = StreamSupport
-      .stream(db.view(CustomType1.class).index("int").spliterator(), false)
-      .map(e -> e.num)
-      .collect(Collectors.toList());
+    try (KVStoreIterator<CustomType1> iterator =
+      db.view(CustomType1.class).index("int").closeableIterator()) {
+      List<Integer> results = StreamSupport
+        .stream(Spliterators.spliteratorUnknownSize(iterator, 0), false)
+        .map(e -> e.num)
+        .collect(Collectors.toList());
 
-    assertEquals(expected, results);
+      assertEquals(expected, results);
+    }
   }
 
   @Test
```

Comment:
This pr clears all similar cases except `testCloseRocksDBIterator` in `LevelDB/RocksDBSuite` because this case seems use to check `finalize` mechanism.

After this pr, we can consider cleaning up `finalize()` related code from `LevelDB/RocksDB` because the `finalize` mechanism is not recommended, and it has been identified as `forRemoval=true` after [Java 18](https://openjdk.java.net/jeps/421)

cc @srowen @HyukjinKwon @dongjoon-hyun